pkg/cloudprovider/providers/azure/auth/azure_auth.go

```diff
@@ -43,29 +43,45 @@ type AzureAuthConfig struct {
 	// The password of the client certificate for an AAD application with RBAC access to talk to Azure RM APIs
 	AADClientCertPassword string `json:"aadClientCertPassword" yaml:"aadClientCertPassword"`
 	// Use managed service identity for the virtual machine to access Azure ARM APIs
-	UseManagedIdentityExtension bool `json:"useManagedIdentityExtension"`
+	UseManagedIdentityExtension bool `json:"useManagedIdentityExtension" yaml:"useManagedIdentityExtension"`
+	// UserAssignedID for explict MSI feature. It only used when UseManagedIdentityExtension is true
+	// If UserAssignedID is empty, it means to use implicit MSI rather than explict MSI
+	UserAssignedID string `json:"userAssignedID" yaml:"userAssignedID"`
 	// The ID of the Azure Subscription that the cluster is deployed in
 	SubscriptionID string `json:"subscriptionId" yaml:"subscriptionId"`
 }
 
 // GetServicePrincipalToken creates a new service principal token based on the configuration
+// There are multiple ways to retrieve access token, they are attempted in the following order
```

Comment:
Expand comment so that users know the first two are only tried when specifically enabled in the config file. Current text sort of sounds like it will always be automatically tried.

Reply:
I will let the code to explain itself. (Removed additional comment)
I was trying to make it easier to understand, not trying to introduce confusion. :)